node/src/blockchain/blockchain_bridge.rs

```diff
@@ -521,7 +531,7 @@ impl BlockchainBridge {
 
     pub fn extract_max_block_count(&self, error: BlockchainError) -> Option<u64> {
         let regex_result =
-            Regex::new(r".* (max:|more than|allowed for your plan:|limited to) (?P<max_block_count>\d+)(.*| blocks at once\.)")
+            Regex::new(r".* (max: |allowed for your plan: |is limited to |block range limit \()(?P<max_block_count>\d+).*")
```

Comment:
This strikes me as probably being narrowly tailored to a particular blockchain client service. Do we only ever use that service in real life, or are there others?

Furthermore, is there a way to tell that an error means the max block count is too big even if we don't get a specific message we can pull apart with a regex to find out what the maximum acceptable size is? If we could identify a range-too-big message (as opposed to an invalid-request message or a my-connection-is-down message or whatever), but the regex failed on it, we could just cut the max block count in half (or down to one-tenth, or whatever) and try again, right?  It seems as though eventually we should get it down to a size that works.

Reply:
Without fawning over Dan, this is an interesting idea. :)